pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go

```diff
@@ -80,6 +94,11 @@ func (a *Authenticator) AuthenticatePassword(username, password string) (user.In
 
 	resp, err := a.client.Do(req)
 	if err != nil {
+		if _, ok := err.(*RedirectAttemptedError); !ok {
+			// If this was a redirect, treat it as a simple
+			// authentication failure, not an error
```

Comment:
@liggitt I guess it comes down to how we handle errors. Right now, if this is returning an actual error, then what the end-user gets back is a 500 error from the `tokenRequest()` call which replaces this error with a different one about refusing to retry. That's not a good user experience for a number of reasons.

To me, a 500 error means that there is a bug in the code somewhere. A 500 error coming out of a *handled* error condition is incorrect. If we want to report a proper error code, we need to decide which one is most appropriate here. At a glance, I'd say it should be one of 404, 410 or 421 (or possibly 418 to be a little tongue-in-cheek).

Thoughts?